### DISCUSSION TITLE
How could I use the request or context in formatError()?

### DISCUSSION THREAD

**Discussion by jleeothon** [graphql/graphql-http#134] [Q&A] [answered]
Hello! I'm migrating from `express-graphql` to `graphql-http`. The function `customFormatErrorFn()` from the former maps naturally to the latter's `formatError()`. However there are some differences in the codebase that I maintain, which make a direct migration a bit hard:

Before:

```js
const middleware = graphqlHTTP(function (request, response, params) {
  return {
    customFormatErrorFn(error) {
      // I have access to request, response, params, including request.myContext
      request.context.logger.error("blabla");
    }
};

app.use('/some/path', middleware);
```

After:

```js
const middleware = createHandler({
  formatError(error) {
    // Have access to error, but not to request or context
  }
});

app.use('/some/path', middleware);
```

In my specific use case, we're calling a logger as e.g. `request.myContext.logger.info()`.

I'm trying to find a workaround with e.g. `onSubscribe()`, or by adding another middleware before or after, but haven't made it work yet.

Any support is appreciated! Thanks!

**Comment by enisdenjo** [comment, marked as answer]
At the moment, this syntax is not supported. This could be a good feature to add though.

However, you can achieve the same thing by using the createHandle IIFE style:

```diff
- const middleware = createHandler({
+ const middleware = (request, response, params) => createHandler({
  formatError(error) {
    // Have access to error and request or context
  }
- });
+ })(request, response, params);

app.use('/some/path', middleware);
```

Since the creation of the handler is a cheap operation, this should not cause any issues performance-wise.

**Reply by jleeothon** [reply]
Thank you. Will try this!
